cmake/Findfirebase_cpp_sdk.cmake

```diff
@@ -0,0 +1,64 @@
+# Copyright 2018 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(FindPackageHandleStandardArgs)
+
+find_path(
+  FIREBASE_CPP_SDK_DIR include/firebase/future.h
+  HINTS
+    $ENV{FIREBASE_CPP_SDK_DIR}
+    ${FIREBASE_BINARY_DIR}/external/src/firebase_cpp_sdk
+)
+
+find_path(
+  FIREBASE_CPP_INCLUDE_DIR firebase/future.h
+  PATHS
+    ${FIREBASE_CPP_SDK_DIR}/include
+)
+
+if(APPLE)
+  if("${CMAKE_OSX_SYSROOT}" MATCHES "iphoneos")
+    # iOS
+    set(FIREBASE_CPP_LIB_DIR ${FIREBASE_CPP_SDK_DIR}/libs/ios/${CMAKE_SYSTEM_PROCESSOR})
+  else()
+    set(FIREBASE_CPP_LIB_DIR ${FIREBASE_CPP_SDK_DIR}/libs/darwin/universal)
+  endif()
+endif()
+
+find_library(
+  FIREBASE_CPP_APP_LIBRARY
+  NAMES firebase_app
+  HINTS
+    ${FIREBASE_CPP_LIB_DIR}
```

Comment:
Though come to think of it, travis might prevent you from submitting with linux in a broken state...

Reply:
Yeah, I think a better patch would be

```
elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
  set(FIREBASE_CPP_LIB_DIR ${FIREBASE_CPP_SDK_DIR}/libs/linux/${CMAKE_SYSTEM_PROCESSOR})
```

The crash comes from the fact that the C++ SDK is compiled with gcc4 which predates gcc5's new C++11 ABI. 

I thought I could fix this by adding something like this to the test CMakeLists.txt:

```diff
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  # Disable C++11 ABI support on Linux. The Firebase C++ SDK is currently built
+  # with gcc 4 which does not yet support the new ABI.
+  #
+  # TODO(mcg): Propagate this to every C++ translation unit, including all
+  # dependencies--even the ExternalProject ones.
+  target_compile_definitions(
+    firebase_firestore_cpp_test
+    PUBLIC
+    _GLIBCXX_USE_CXX11_ABI=0
+  )
+endif()
```

Unfortunately, this now fails to build because gtest has been compiled with the C++11 ABI but the test iteself has not. The way to fix this is to propagate this down to all of the dependencies we build via ExternalProject. I'm working on that now.